storage/innobase/xtrabackup/test/inc/common.sh

```diff
@@ -1150,5 +1150,27 @@ function backup_and_restore() {
   verify_db_state $db
 }
 
+### kill all running queries
+function kill_all_queries()
+{
+  run_cmd $MYSQL $MYSQL_ARGS --force --batch  test <<EOF
+  select concat('KILL ',id,';') from information_schema.processlist
+  where user='root' and time > 1 into outfile '$MYSQLD_TMPDIR/killall.sql';
+  source $MYSQLD_TMPDIR/killall.sql;
+EOF
+  rm -f $MYSQLD_TMPDIR/killall.sql
+}
+
+function wait_for_process_to_start() {
+  local pid=$1
+  vlog "wait for $pid to start "
+  while  ! kill -0 $pid >/dev/null 2>&1
+  do
+    vlog "waiting for pid $pid to start"
+    sleep 1
+  done
+}
```

Comment:
I don't think this loop is necessary. Check existing kill_long_selects.sh which use "wait $pid". There is an existing function. Please use that and move to .inc file

Reply:
sorry I didn't understand . can you please tell me exact function?

Reply:
bg_wait_ok

Reply:
IIUC bg_wait_ok is just wait untill the process is finish. but it is the background process and we wait  until it is started.

Reply:
Hmm, if you have pid, then process is already started right? why do u need this hack?